mlflow/store/artifact/utils/models.py

```diff
@@ -63,28 +57,32 @@ def _parse_model_uri(uri):
     if parsed.scheme != "models":
         raise MlflowException(_improper_model_uri_msg(uri))
     path = parsed.path
-    m = _MODEL_URI_REGEX.match(path)
-    if m is None:
+    if not path.startswith("/") or len(path) <= 1:
         raise MlflowException(_improper_model_uri_msg(uri))
-    gd = m.groupdict()
-    model_name = gd.get("model_name")
-    suffix = gd.get("suffix")
-    alias = gd.get("alias")
-    if (model_name.strip() == "") or (suffix and alias) or (suffix is None and alias is None):
+
+    parts = path[1:].split("/")
+    if len(parts) > 2 or parts[0].strip() == "":
         raise MlflowException(_improper_model_uri_msg(uri))
 
-    if alias:
-        # The URI is an alias URI, e.g. "models:/AdsModel1@Champion"
-        return ParsedModelUri(model_name, alias=alias)
-    elif suffix.isdigit():
-        # The suffix is a specific version, e.g. "models:/AdsModel1/123"
-        return ParsedModelUri(model_name, version=suffix)
-    elif suffix.lower() == _MODELS_URI_SUFFIX_LATEST.lower():
-        # The suffix is the 'latest' string (case insensitive), e.g. "models:/AdsModel1/latest"
-        return ParsedModelUri(model_name)
+    if len(parts) == 2:
+        if parts[1].strip() == "":
+            raise MlflowException(_improper_model_uri_msg(uri))
+        # The URI is in the suffix format
+        if parts[1].isdigit():
+            # The suffix is a specific version, e.g. "models:/AdsModel1/123"
+            return ParsedModelUri(parts[0], version=parts[1])
+        elif parts[1].lower() == _MODELS_URI_SUFFIX_LATEST.lower():
+            # The suffix is the 'latest' string (case insensitive), e.g. "models:/AdsModel1/latest"
+            return ParsedModelUri(parts[0])
+        else:
+            # The suffix is a specific stage (case insensitive), e.g. "models:/AdsModel1/Production"
+            return ParsedModelUri(parts[0], stage=parts[1])
     else:
-        # The suffix is a specific stage (case insensitive), e.g. "models:/AdsModel1/Production"
-        return ParsedModelUri(model_name, stage=suffix)
+        # The URI is an alias URI, e.g. "models:/AdsModel1@Champion"
+        alias_parts = parts[0].split("@")
```

Comment:
What happens if your model name is `foo@bar`?

Reply:
then you're either doing `foo@bar@alias` or `foo@bar/version`, so it doesn't matter :)

Reply:
What if you happen to have a model with a name `foo` and an alias `bar`, and use `models:/foo@bar` by mistake instead of `models:/foo@bar/suffix` or `models:/foo@bar@baz`? Would that result in loading an unintended model?

Reply:
Offline discussion: this is a user error, we don't need to worry about this.